rtengine/iplocallab.cc

```diff
@@ -20521,6 +20536,37 @@ void ImProcFunctions::Lab_Local(
             }
         }
     }
+    //verify that RGB values are > 0.f issue 7121 to avoid crash
```

Comment:
Which part of the code is crashing if values are not positive? Shouldn't that part of the code clip the RGB values? That would prevent crashes caused by other tools that we do not know about yet.

Reply:
I checked possible crash cases with negative values. 

There is of course Laplacians, but also RGB curves (Tab Color main).

There were also, but this is handled calls to workingtrc  (Abstract Profile and Selective editings).

I think it is "better" not to clip the negative RGB values, knowing that with Prophoto as a "working profile" we are in imaginary colors. But since the user can change the working profile, it's better this way.

If new crashes appear (with negative values) then these cases will have to be added gradually: which function causes the crash and therefore activate the variable "notzero".

Additionally, to save time and memory, I only call functions if necessary.

Thank you

Jacques

Reply:
Ok. Where was the crash happening? I'm not able to reproduce the crash, but I'm curious.